crates/storage/provider/src/traits/block.rs

```diff
@@ -32,6 +32,13 @@ pub trait BlockExecutionReader: BlockReader + Send + Sync {
     ) -> ProviderResult<Chain>;
 }
 
+/// This just receives state, or [`ExecutionOutcome`], from the provider
+#[auto_impl::auto_impl(&, Arc, Box)]
+pub trait StateReader: Send + Sync {
```

Comment:
do we want a new trait for this or can this be integrated in BlockExecutionReader ?

Reply:
I was planning on adding this to `BlockExecutionReader`, but only needed / implemented the method for a single block. We should merge the two traits, I'll create an issue